server/repository/assessment/assessment/getSectionMetaData.ts

```diff
@@ -49,14 +51,16 @@ export const getSectionMetaData = async (
                jsonb_agg(t.* order by t.id) as tables
         from "tables" t
                  left join ts on t.table_section_id = ts.id
-        group by ts.id, ts.uuid, ts.props, ts.section_id;
+        group by ts.id, ts.uuid, ts.props, ts.section_id, ts.section_name;
```

Comment:
I had some issues optimizing this query to return the above todo
>   // TODO: This query should be optimized to return Record<[sectionName], Array<TableSection>>

but code on frontend etc are updated to support the final result

Reply:
```sql

with "row" as (
    select s.props ->> 'name' as section_name,
           to_jsonb(ts.*)     as table_section,
           to_jsonb(t.*)      as "table",
           jsonb_set(
                   to_jsonb(r.*),
                   '{"cols"}',
                   jsonb_agg(c.* order by c.id)
               )              as row
    from assessment_fra.col c
             left join assessment_fra.row r on r.id = c.row_id
             left join assessment_fra."table" t on t.id = r.table_id
             left join assessment_fra.table_section ts on ts.id = t.table_section_id
             left join assessment_fra.section s on ts.section_id = s.id
--     where s.props ->> 'name' in ('extentOfForest')
--   and ts.props -> 'cycles' ? $2
--   and t.props -> 'cycles' ? $2
--   and r.props -> 'cycles' ? $2
--   and c.props -> 'cycles' ? $2
    group by s.props ->> 'name', to_jsonb(ts.*), to_jsonb(t.*), to_jsonb(r.*)
),
     "table" as (
         select r.section_name,
                r.table_section,
                jsonb_set(
                        r."table",
                        '{"rows"}',
                        jsonb_agg(r.row order by (r.row ->> 'id')::numeric)
                    ) as "table"
         from row r
         group by r.section_name, r.table_section, r."table"
     ),
     table_section as (
         select t.section_name,
                jsonb_set(
                        t.table_section,
                        '{"tables"}',
                        jsonb_agg(t."table" order by (t."table" ->> 'id')::numeric)
                    ) as table_section
         from "table" t
         group by t.section_name, t.table_section
     ),
     section as (
         select ts.section_name, jsonb_agg(ts.table_section) as table_sections
         from table_section ts
         group by ts.section_name
     )
select jsonb_object_agg(s.section_name, s.table_sections) as data
from section s
;
```